server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java

```diff
@@ -50,9 +50,11 @@
 public class QProfileFactory {
 
   private final DbClient db;
+  private final UuidFactory uuidFactory;
 
-  public QProfileFactory(DbClient db) {
+  public QProfileFactory(DbClient db, UuidFactory uuidFactory) {
```

Comment:
The examples of responses for api/qualityprofiles should be updated accordingly. They are for example `"key" : "my-new-profile-07284"` in actions `create`, `compare`, `inheritance` and `search`.